src/components/AssetSelection/components/AssetChainDropdown/AssetChainDropdown.tsx

```diff
@@ -0,0 +1,89 @@
+import type { ButtonProps, MenuProps } from '@chakra-ui/react'
+import {
+  Button,
+  Menu,
+  MenuButton,
+  MenuItemOption,
+  MenuList,
+  MenuOptionGroup,
+  Tooltip,
+} from '@chakra-ui/react'
+import type { AssetId } from '@shapeshiftoss/caip'
+import { useCallback, useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { isAssetSupportedByWallet } from 'state/slices/portfolioSlice/utils'
+import {
+  selectAssetChainNameById,
+  selectRelatedAssetIdsInclusiveSorted,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { AssetRowLoading } from '../AssetRowLoading'
+import { AssetChainRow } from './AssetChainRow'
+
+type AssetChainDropdownProps = {
+  assetId?: AssetId
+  onClick: (arg: AssetId | undefined) => void
+  buttonProps?: ButtonProps
+  isLoading?: boolean
+  isError?: boolean
+} & Omit<MenuProps, 'children'>
+
+export const AssetChainDropdown: React.FC<AssetChainDropdownProps> = ({
+  assetId,
+  onClick,
+  buttonProps,
+  isLoading,
+  isError,
+  ...menuProps
+}) => {
+  const {
+    state: { wallet },
+  } = useWallet()
+  const translate = useTranslate()
+  const assetChainName = useAppSelector(state => selectAssetChainNameById(state, assetId ?? ''))
```

Comment:
Would use `displayName` or `networkName` vernacular for grepping purposes here - we don't really use `chainName` anywhere to refer to this

Reply:
👍 landed on `chainDisplayName` here f9ef7bf0243e7c0591b22bcf3dce7c436e97a102